compiler/plc_xml/src/xml_parser/control.rs

```diff
@@ -0,0 +1,99 @@
+use ast::ast::{AstFactory, AstNode, Operator};
+use plc_diagnostics::diagnostics::Diagnostic;
+
+use crate::model::{
+    control::{Control, ControlKind},
+    fbd::{Node, NodeIndex},
+};
+
+use super::ParseSession;
+
+impl Control {
+    pub(crate) fn transform(&self, session: &ParseSession, index: &NodeIndex) -> Result<AstNode, Diagnostic> {
+        match self.kind {
+            ControlKind::Jump => unimplemented!(),
+            ControlKind::Label => unimplemented!(),
+            ControlKind::Return => transform_return(self, session, index),
+        }
+    }
+}
+
+fn transform_return(
+    control: &Control,
+    session: &ParseSession,
+    index: &NodeIndex,
+) -> Result<AstNode, Diagnostic> {
+    let Some(ref_local_id) = control.ref_local_id else {
+        let location = session.range_factory.create_block_location(control.local_id, None);
+        return Err(Diagnostic::empty_control_statement(location));
+    };
+
+    let Some(node) = index.get(&ref_local_id) else {
+        let location = session.range_factory.create_block_location(ref_local_id, None);
```

Comment:
Yeah, I was thinking of console errors which the user _might_ see but your suggestion makes a lot more sense!

Reply:
I missed this at first but saw it again while implementing the jumps. I think this style of diagnostics belongs to the validator.
It is not a syntax error that should stop the xml parsing, it's just a missing statement. We should get the validator to say i'm missing a connection here. We can probably do this by making the ReturnStatement always get a TRUE value on the ST level, or use a different block type (ConditionalReturn instead of Return)

Reply:
So i was having more thoughts on this, maybe these are parse errors (unconnected return) but we should not break parsing, just use diagnostics so it should go in the CFC Diagnoctics issue